tools/mo/openvino/tools/mo/utils/ir_reader/internal_ops/squeeze.py

```diff
@@ -3,12 +3,23 @@
 
 from openvino.tools.mo.graph.graph import Node
 from openvino.tools.mo.ops.squeeze import Squeeze
+from openvino.tools.mo.front.common.partial_infer.utils import shape_array, is_fully_defined
 
 
 class SqueezeInternal(Squeeze):
     @staticmethod
     def infer(node: Node):
-        axis_value = node.in_port(1).data.get_value()
-        Squeeze.infer(node)
-        # preserve initial axis value
-        node.in_port(1).data.set_value(axis_value)
+        if node.is_in_port_connected(1):
+            axis_value = node.in_port(1).data.get_value()
+            Squeeze.infer(node)
+            # preserve initial axis value
+            node.in_port(1).data.set_value(axis_value)
+        else:
+            # Squeeze without axes provided
```

Comment:
Am I correct that TF FE is a source of such IRs with `Squeeze` nodes without axis? And we need to avoid such `Squeeze` nodes. What is downside of such IRs? May be we need to introduce new `Squeeze` specification with this limitation?

Reply:
Downside of such IRs is in dynamic cases. If input shape is not known the output rank is not known.

Reply:
May be we should consider to have the second input as mandatory?

Reply:
I don't think that would be a good idea. I suppose it was introduced as optional, because some frameworks might have it as optional. So to comply with such frameworks we better have it optional too. But avoid this in frontends if possible

Reply:
I see that ONNX needs such feature.